### PR TITLE
[search_quality] Add num_threads parameter to features_collector_tool.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

File: search/search_quality/features_collector_tool/features_collector_tool.cpp [hunk 1]
```diff
@@ -156,7 +157,35 @@ int main(int argc, char * argv[])
     TestSearchRequest request(*engine, params);
     request.Run();
 
-    auto const & results = request.Results();
+    savedResults[i] = request.Results();
+  };
+
+  auto const samplesPerThread = static_cast<size_t>(ceil(samples.size() / FLAGS_num_threads));
```
Comment (mpimenov) [inline search/search_quality/features_collector_tool/features_collector_tool.cpp]:
```
auto const samplesPerThread = (samples.size() + FLAGS_num_threads - 1) / FLAGS_num_threads;
```

File: search/search_quality/features_collector_tool/features_collector_tool.cpp [hunk 2]
```diff
@@ -156,7 +157,35 @@ int main(int argc, char * argv[])
...
+  auto const getFirstSampleId = [&](size_t n) {
+    return samplesPerThread * n;
+  };
+  auto const getLastSampleId = [&](size_t n) {
+    return min(samplesPerThread * (n + 1) - 1, samples.size() - 1);
+  };
+
+  auto const fn = [&](size_t threadNumber) {
+    for (auto i = getFirstSampleId(threadNumber); i <= getLastSampleId(threadNumber); ++i)
```
Comment (mpimenov) [inline search/search_quality/features_collector_tool/features_collector_tool.cpp]:
```
auto const beg = samples.size() * threadNumber / numThreads;
auto const end = samples.size() * (threadNumber + 1) / numThreads;
for (auto i = beg; i < end; ++i)
  ...
```
?

File: search/search_quality/features_collector_tool/features_collector_tool.cpp [hunk 3]
```diff
@@ -156,7 +157,35 @@ int main(int argc, char * argv[])
...
+      process(i);
+  };
+
+  {
+    base::thread_pool::computational::ThreadPool pool(FLAGS_num_threads);
```
Comment (mpimenov) [inline search/search_quality/features_collector_tool/features_collector_tool.cpp]: Why do you need pool here? search::Engine should take care of this automatically, shouldn't it?

Comment (tatiana-yan) [reply search/search_quality/features_collector_tool/features_collector_tool.cpp]: только если TestSearchRequest переписать

**Comment by tatiana-yan** [comment]
@mpimenov PTAL

**Pull request merged**